data_and_models/pipelines/ner/dvc.yaml

```diff
@@ -1,404 +1,170 @@
-    - Dockerfile
-    - config.cfg
-    - initialize.py
-    - ../../annotations/ner/annotations6_EmmanuelleLogette_2020-07-07_raw4_TaxonChebi.train.spacy
-    - ../../annotations/ner/annotations6_EmmanuelleLogette_2020-07-07_raw4_TaxonChebi.dev.spacy
-    outs:
-    - ../../models/ner/model3/model-best/
-  train_model_4:
-    cmd: >-
-      python -m spacy train
-      config.cfg
-      --output ../../models/ner/model4/
-      --code initialize.py
-      --paths.train ../../annotations/ner/annotations9_EmmanuelleLogette_2020-07-08_raw6_CelltypeProtein.train.spacy
-      --paths.dev ../../annotations/ner/annotations9_EmmanuelleLogette_2020-07-08_raw6_CelltypeProtein.dev.spacy
-      --paths.base_model en_ner_jnlpba_md
-      &&
-      python patch.py ../../models/ner/model4/model-best/vocab/strings.json
-    deps:
-    - Dockerfile
-    - config.cfg
-    - initialize.py
-    - ../../annotations/ner/annotations9_EmmanuelleLogette_2020-07-08_raw6_CelltypeProtein.train.spacy
-    - ../../annotations/ner/annotations9_EmmanuelleLogette_2020-07-08_raw6_CelltypeProtein.dev.spacy
-    outs:
-    - ../../models/ner/model4/model-best/
-  train_model_5:
-    cmd: >-
-      python -m spacy train
-      config.cfg
-      --output ../../models/ner/model5/
-      --code initialize.py
-      --paths.train ../../annotations/ner/annotations15_EmmanuelleLogette_2020-09-22_raw9_Pathway.train.spacy
-      --paths.dev ../../annotations/ner/annotations15_EmmanuelleLogette_2020-09-22_raw9_Pathway.dev.spacy
-      --paths.base_model en_ner_craft_md
-      &&
-      python patch.py ../../models/ner/model5/model-best/vocab/strings.json
-    deps:
-    - Dockerfile
-    - config.cfg
-    - initialize.py
-    - ../../annotations/ner/annotations15_EmmanuelleLogette_2020-09-22_raw9_Pathway.train.spacy
-    - ../../annotations/ner/annotations15_EmmanuelleLogette_2020-09-22_raw9_Pathway.dev.spacy
-    outs:
-    - ../../models/ner/model5/model-best/
+  convert_annotations:
+    foreach:
+      annotation5:
+        annotation_filename: annotations5_EmmanuelleLogette_2020-06-30_raw2_Disease
+      annotation6:
+        annotation_filename: annotations6_EmmanuelleLogette_2020-07-07_raw4_TaxonChebi
+      annotation9:
+        annotation_filename: annotations9_EmmanuelleLogette_2020-07-08_raw6_CelltypeProtein
+      annotation10:
+        annotation_filename: annotations10_EmmanuelleLogette_2020-08-28_raw1_raw5_10EntityTypes
+      annotation11:
+        annotation_filename: annotations11_CharlotteLorin_2020-08-28_raw1_10EntityTypes
+      annotation12:
+        annotation_filename: annotations12_EmmanuelleLogette_2020-08-28_raw7_10EntityTypes
+      annotation13:
+        annotation_filename: annotations13_CharlotteLorin_2020-09-02_raw7_10EntityTypes
+      annotation14:
+        annotation_filename: annotations14_EmmanuelleLogette_2020-09-02_raw8_CellCompartmentDrugOrgan
+      annotation15:
+        annotation_filename: annotations15_EmmanuelleLogette_2020-09-22_raw9_Pathway
```

Comment:
I'm wondering if the names `annotationXX` are used anywhere? Are they used at all? On the [DVC documentation](https://dvc.org/doc/user-guide/project-structure/pipelines-files#foreach-stages) I read that it's possible to write something like
```yaml
stages:
  convert_annotations:
    foreach:
      - annotations5_EmmanuelleLogette_2020-06-30_raw2_Disease
      - annotations6_EmmanuelleLogette_2020-07-07_raw4_TaxonChebi
      - ...
    do:
      cmd: python preprocess.py ../../annotations/ner/${item}.jsonl
      ...
```
Would we break anything if we used this syntax?

Reply:
That is a really good point. This only thing is that the naming in the `dvc.lock` is going to become `convert_annotations@0`, `convert_annotations@1`, ... but it is not really a problem.

Reply:
Done in 60cdbde